src/ClisBnbStrategy.sol

```diff
@@ -0,0 +1,331 @@
+
+        // if the vault balance is less than the assets to withdraw, unstake the slisBnb
+        if (vaultBalance < assets) {
+            uint256 amountToUnstake = assets - vaultBalance;
+            // unstake the slisBnb
+            slisBnbProvider().release(address(this), amountToUnstake);
+            // check if the amount of slisBnb received is less than the amount to unstake
+            uint256 slisBnbAmountReceived = slisBnb().balanceOf(address(this)) - vaultBalance;
+            // if the amount of slisBnb received is less than the amount to unstake, revert
+            // this can be possible if withdrawal is changed from instant to delayed by Lista for slisBnb
+            if (slisBnbAmountReceived < amountToUnstake) {
+                revert InsufficientSlisBnbReceived();
+            }
+        }
+
+        // transfer the assets to the receiver
+        SafeERC20.safeTransfer(IERC20(asset_), receiver, assets);
+
+        emit WithdrawAsset(caller, receiver, owner, asset_, assets, shares);
+    }
+
+    /**
+     * @notice Internal function to get the available amount of assets.
+     * @dev This function is used for slisBnb asset
+     * @param asset_ The address of the asset.
+     * @return availableAssets The available amount of assets.
+     */
+    function _availableAssets(address asset_) internal view virtual override returns (uint256 availableAssets) {
+        IERC20 _slisBnb = slisBnb();
+        if (asset_ == address(_slisBnb)) {
+            availableAssets = _slisBnb.balanceOf(address(this));
+            // add the locked slisBnb to the available assets as it can be 1:1 claimable by this strategy from Lista
+            availableAssets += listaInteraction().locked(asset_, address(this));
+        } else {
+            availableAssets = super._availableAssets(asset_);
+        }
+    }
+
+    /**
+     * @notice Computes the total assets in the vault.
+     * @dev Since this strategy only supports slisBnb, this function will return the total balance of slisBnb managed by the vault.
+     * @return totalBaseBalance The total assets in the vault in slisBnb.
+     */
+    function computeTotalAssets() public view virtual override returns (uint256 totalBaseBalance) {
+        totalBaseBalance = super.computeTotalAssets();
+        // get the slisBnb address
+        IERC20 _slisBnb = slisBnb();
+        // get the locked slisBnb by this strategy from Lista
+        // @dev the amount of slisBnb present in vault is already included in totalBaseBalance during call to super.computeTotalAssets()
+        totalBaseBalance += listaInteraction().locked(address(_slisBnb), address(this));
+    }
+
+    /**
+     * @notice Sets the lista interaction contract address.
+     * @param _listaInteraction The address of the lista interaction contract.
+     */
+    function setListaInteraction(Interaction _listaInteraction) external onlyRole(LISTA_DEPENDENCY_MANAGER_ROLE) {
```

Comment:
shouldn't we just set this and slisBnb in the initialize and not have these separate setters?

i think that's cleaner since they shouldn't change

the slisBNB address is 100% guaranteed to not change so i would not have these setters for ListaInteraction and slisBNB here since they are not needed, to keep it simple

Reply:
same for slisBNBProvider i would say, unless we know it should change, but very unlikely i think